docs/input.rst

```diff
@@ -141,7 +141,30 @@ still provide the following information:
   ``'NOT_ALIVE_DISPOSED'``), the sample data contains the value of the key that
   has been disposed. You can access the key fields only. See
   :ref:`Accessing key values of disposed samples`.
-  
+
+Returning data and meta-data samples
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+The methods :meth:`Input.take()` and :meth:`Input.read()` modify the internal state
```

Comment:
I would suggest edits to make the documentation more relevant to the user:

The methods ... make the data available through the Input.samples collection. This data is available until ...

The data in Input.sample uses internal resources that in some situations may need to be returned sooner so that new data can be received. To do that, you can call Input.wait with... or explicitly call Input.return_samples at any moment.